static/js/lib/ckeditor/plugins/MarkdownListSyntax.ts

```diff
@@ -0,0 +1,52 @@
+import { ShowdownExtension } from "showdown"
+import { TurndownRule } from "../../../types/ckeditor_markdown"
+import MarkdownSyntaxPlugin from "./MarkdownSyntaxPlugin"
+
+/**
+ * This "extension" is addresses an incompatibility between
+ *  - the HTML that showdown produces for lists
+ *  - the HTML that CKEditor expects for lists
+ *
+ * When a markdown list item contains multiple lines, showdown puts <p> tags
+ * around all list items (and multipple <p> tags for the multi-line list item).
+ * See https://github.com/showdownjs/showdown/wiki/Showdown's-Markdown-syntax#known-differences-and-gotchas.
+ *
+ * CKEditor, on the other hand, forbids paragraphs inside list items. Both are
+ * "block" elements in its schema; see https://ckeditor.com/docs/ckeditor5/latest/framework/guides/deep-dive/schema.html#defining-additional-semantics
+ * and block elements cannot contain other block elements. (There is discussion
+ * of having a list variant that can contain block elements; see
+ * https://github.com/ckeditor/ckeditor5/issues/2973 for more.)
+ *
+ * So here we add a Showdown rule to replace list item paragraph tags with
+ * linebreaks, the same as CKEditor produces.
+ *
+ */
+export default class MarkdownListSyntax extends MarkdownSyntaxPlugin {
```

Comment:
@alicewriteswrongs You mentioned a few possibilities in slack
> but anyway - we can make an intervention at a few points, [1] we could try to fix how Showdown handles this, or [2] we could try to edit and fix up it’s html output before it’s passed to ckeditor, or [3] we could try to see if there’s a way to make ckeditor tolerate <p> tags inside of list items

Since this is an "output" showdown rule, it basically seems like [2] to me. Using an output rule seemed more convenient than a "lang" rule since having the html tree is convenient for locating `li > p`. 

I did not explore (3) _too_ much beyond reading about the schema, and block elements not containing block elements and reading through some of https://github.com/ckeditor/ckeditor5/issues/2973.